ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -174,7 +174,7 @@ public MaterializedWindowedTable windowed() {
       final Builder<WindowedRow> builder = ImmutableList.builder();
 
       for (final WindowedRow row : result) {
-        filterAndTransform(row.windowedKey(), row.value(), row.rowTime())
+        filterAndTransform(row.windowedKey(), getIntermediateRow(row), row.rowTime())
```

Comment:
Unsure if we even need to append the extra columns for windowed rows. Windowed state stores might need more testing

Reply:
Could you elaborate a bit more here? Is that because the windowed key may have the needed columns for filtering?

Reply:
@guozhangwang my understanding is that windowed tables always need an aggregation, so they have a state store backing the aggregated table. If that is the case, then we don't need to append this extra metadata as `KsqlMaterialization` is sophisticated enough to handle those cases (we have test cases for that). I noticed that we follow a similar pattern in `ProjectOperator` and `SelectOperator` of generating intermediate rows on which filters and transformations can be applied. I added these fields as a hedge against potential cases that we might miss (obviously it comes at a cost). That being said, I have a couple of questions for you:
1) Are there any type of Windowed tables that are not queryable today that we want to be able to query?
2) Can Windowed tables be derived without doing any aggregations? (Specifically, `GROUP BY` aggregations)

Reply:
I haven't tested it by try a windowed query with group by and having clause.  I think that might trigger `KsqlMaterialization`.  In that case, presumably you could say `HAVING WINDOWSTART > 20` and therefore need to have those columns.

Reply:
+1 to @AlanConfluent , I think today if you have a `WINDOW BY + GROUP BY + HAVING` you would have a first windowed table from `windowBy+groupBy` aggregations, and then a second windowed table from `having` as a filtering condition. So in that sense not all windowed tables should be generated with aggregations, they can also be generated from table statless operators from other existing windowed tables.

Reply:
It would be good to have this as a RQTT test where the having clause mentions windowstart or windowend